### PR TITLE
Add empty state help for projects page

### PR DESCRIPTION
**Pull request by spadgett**
Adds some help text when you first login to OpenShift and have no projects.

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres @smarterclayton @liggitt @ajacobs21e what do you think?

Here is an empty project for reference. We might change that message because we're repeating ourselves.

openshift_web_console

**Comment by spadgett** [comment]
I wanted to add an OpenShift logo beside the "Welcome to OpenShift" header, but I couldn't find one that worked on a gray background.

**Comment by jwforres** [comment]
Couple comments:
* in v3 we dont refer to them as QuickStarts, we have been calling them instant apps
* don't say "community" documentation since this will be in enterprise as well and enterprise would link to the enterprise docs
* I think we should still have the same type of help when you can't create projects, just change the "create a project" references as needed to say request a project or something

@sg00dwin see sam's comment about wanting an openshift logo https://github.com/openshift/origin/pull/4317#issuecomment-133524532

**Comment by spadgett** [comment]
We should probably open a doc defect since it uses the term QuickStart:

https://docs.openshift.org/latest/admin_guide/install/first_steps.html#creating-quickstart-templates

**Comment by liggitt** [comment]
careful with logos or community references... those all need to be swappable in alternate releases

File: assets/app/views/projects.html [hunk 1]
```diff
@@ -1,28 +1,94 @@
...
+        needed.
+      </p>
+
+      <a href="/createProject" class="lead">
+        Create your first project
+      </a>
+    </div>
+
+    <div ng-if="!canCreate" class="empty-state-msg">
+      <p class="lead">Unfortunately, you don't have authority to create projects.</p>
```
Comment (liggitt) [inline assets/app/views/projects.html]: remember, `newProjectMessage` can contain the customized message from the admin. They might have enabled a way for users to create projects for themselves, so we have to show it if it exists (using `linky`), and we shouldn't pair it with an "Unfortunately..." header

**Comment by jwforres** [comment]
Ok I wasn't aware of that, I guess we started calling them QuickStarts
again? @smarterclayton ? @bparees?

**Comment by bparees** [comment]
@jwforres i'm not sure we ever did a great job of picking one or the other.

v2 quickstart:  a way to get multiple cartridges created at once.
v2 instant-app: a way to get an app created from a specific source repo.  may also include multiple cartridges, so arguably an enhanced quickstart

v3 template: can serve in the role of either a quickstart or an instant-app (might provide a default source-uri to a useful app, or might expect you to provide a source-uri because it's just a generic helper for building java apps with a database)

but i'm ok with changing the docs to s/QuickStart/InstantApp/ if that's what we want.

**Comment by smarterclayton** [comment]
I'm not in love with the white box.  I would prefer something more subtle.  To many overlapping rectangles.

**Comment by jwforres** [comment]
which white box... the one on the empty project list is exactly what we do
in v2.  Are you talking about the one after you are in a project and the
project is empty?

**Comment by smarterclayton** [comment]
Yes, project is empty.  Basically when it's white left nav and white top bar, then gray, then white box, it looks out of balance.  I think empty should look empty.

**Comment by liggitt** [comment]
I like it. Empty grey feels sad. Sad is bad.

**Comment by smarterclayton** [comment]
In that first image though, what does it look like when you don't have the white box on the empty projects page?  Flush left alignment with no box border.

**Comment by smarterclayton** [comment]
Or instead of the centered box, do the full width page white bar (like what we have on the homepage and other pages).

**Comment by spadgett** [comment]
Updated for the comments from @jwforres and @liggitt. Using Instant App for now, but I can change it back.

@smarterclayton working on some screens of your suggestions

**Comment by spadgett** [comment]
@smarterclayton Gray background. Maybe too many different fonts.

openshift_web_console

**Comment by liggitt** [comment]
the big link feels unbalanced... it looks as big as the header. Should it be in the "Create a project" portion of the outline?

**Comment by smarterclayton** [comment]
That's a lot of fonts.

**Comment by spadgett** [comment]
White edge to edge background.

openshift_web_console

I can remove some of the fonts.

**Comment by spadgett** [comment]
Fewer fonts:

openshift_web_console

**Comment by smarterclayton** [comment]
The center content would look better with two columns - sentences that wrap the page are unreadable for most humans.  This really needs to be a flow from 1 to 2a and 2b - I would organize them visually that way.

**Comment by smarterclayton** [comment]
Or even make the right column for the center be a picture so it's narrower text.  And cut out some of the text.

**Comment by spadgett** [comment]
Yeah, typically we add generous margins for wide screen widths, although my hack to try edge to edge white removed them. I think the columns are a good ideas, but I'd probably need to stack them on mobile.

**Comment by smarterclayton** [comment]
Stack for sure - a picture would be better even.

On Fri, Aug 21, 2015 at 3:55 PM, Sam Padgett <notifications@github.com>
wrote:

> Yeah, typically we add generous margins for wide screen widths, although
> my hack to try edge to edge white removed them. I think the columns are a
> good ideas, but I'd probably need to stack them on mobile.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4317#issuecomment-133544862>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

**Comment by spadgett** [comment]
@smarterclayton I wouldn't use this image, but is this what you had in mind?

openshift_web_console

**Comment by smarterclayton** [comment]
Yes

**Comment by smarterclayton** [comment]
Bonus points for having a graphic with arrows that shows source code going into a box and then the box being sent out on a conveyor line (or carried by helicopters into a cloud)

**Comment by jwforres** [comment]
cc @sg00dwin

**Comment by spadgett** [comment]
The problem with the edge-to-edge white is the long lines... It looks strange with wide margins (see screenshot). I think it'll still be a problem even with an image unless it's very large.

openshift_web_console

**Comment by ajacobs21e** [comment]
I have two suggestions:
empty state
With minor changes to the messaging - whatever messaging we use should be reviewed by a content writer.

The info is good but it may be more useful if split up onto the relevant pages:
empty state2

empty state3


The white stripe on the background looks strange and doesn't fit with the OpenShift or Patternfly theming.

**Comment by liggitt** [comment]
+1 for splitting the content into multiple pages to just show the next relevant step

**Comment by spadgett** [comment]
OK, starting to split up the content.

I'm not sold on gray without the box. It feels too empty. Maybe if vertically centered, but honestly the gray is kind of dull.

openshift_web_console

Haven't really changed the styles here (yet), but changed the wording.

openshift_web_console

We tried to be careful with the message before because the project might not really be new or empty, even if we didn't show anything in the overview.

**Comment by ajacobs21e** [comment]
The empty state is supposed to look empty to make you want to click that button and fill it in with pretty stuff. 

I'm good with taking out "new" project. Is there a content writer we work with for OpenShift?

**Comment by smarterclayton** [comment]
I'm a little concerned with the white box because we use it to mean "this is a thing in the system you interact with".  The empty state is not quite the same thing.  It sounds more to me like we're lacking the visual language to identify a "take me places, get me started".  And centering it without the box leaves it a bit lost and alone on the page doesn't help strengthen that message.  

Why use small font styles?  If you have 100 or so words to use, and they are important, why not put them into two or three lines together at an H2 level font size?  The suggestion of images was really to create and fill space that you can balance the text against.

All that being said, anything is better than nothing right now, so I'm fine with doing something to get it moving.  We can always fine tune later or have someone spend some time crafting a more impressive visual presence.

**Comment by spadgett** [comment]
Project page without the white box for comparison. This one doesn't feel so empty to me. I can play with font sizes and spacing for the empty project list.

openshift_web_console

**Comment by sg00dwin** [comment]
Recommend removing the h1 heading with line, to pull things up.

**Comment by smarterclayton** [comment]
Left margin is pretty ragged - that's a case where a left image might serve to give separation from the left nav without requiring extra constructs.  Either way, not a huge deal.

**Comment by spadgett** [comment]
More top margin on the all projects page:

screen shot 2015-08-24 at 1 21 20 pm

Mobile:

openshift_web_console

**Comment by jwforres** [comment]
I just want to make sure we don't lose the requirement that we explain what a project is, that was the complaint we had.

**Comment by sg00dwin** [comment]
If the project is empty, then it would seem we shouldn't show the left bar primary nav items.

**Comment by spadgett** [comment]
@sg00dwin There might actually be content there, though. For instance, we filter some pods on the overview, but they might appear in the browse pods page. That's why I want to be a little careful with the language.

Settings might have content like quotas, too.

**Comment by jwforres** [comment]
I disagree from the perspective that a project admin might create a
project, go to settings / members to give a bunch of people access, but
never want to add anything to the project themselves.

**Comment by spadgett** [comment]
Removed `h1` and left-aligned content.

openshift_web_console

**Comment by spadgett** [comment]
@jwforres opinion on the icons?

openshift_web_console

**Comment by spadgett** [comment]
And w/o the negative margin:

openshift_web_console

**Comment by spadgett** [comment]
OK last one:

openshift_web_console

**Comment by spadgett** [comment]
Unrelated travis failure:

```
--- FAIL: TestRequestExecuteRemoteCommand-2 (0.11s)
	remotecommand_test.go:112: 2: stdin: expected "a", got ""
FAIL
```

**Comment by ajacobs21e** [comment]
@spadgett out of the three options I like #1 the best (icons past left margin) however I don't think the icons add anything - unless we also use these same icons on the create flow.

**Comment by jwforres** [comment]
LGTM, we'll follow up with a picture for the initial welcome page when we actually have one to use

last chance for any arguments

**Comment by jwforres** [comment]
@ajacobs21e we use the thunderbolt on templates that don't specify their own icon

**Comment by ajacobs21e** [comment]
@jwforres ok cool. what do you think about using the icons on the initial Create page with the 'From Source' and 'From Template; headings?

**Comment by jwforres** [comment]
@ajacobs21e makes sense to me, would like to see how it looks with them there

**Comment by spadgett** [comment]
openshift_web_console

**Comment by spadgett** [comment]
@jwforres Pushed the icon changes to create.html and fixed the icon widths. PTAL

openshift_web_console

**Comment by jwforres** [comment]
[merge]

**Comment by smarterclayton** [comment]
That merge looked like it had some angst behind it :)

On Tue, Aug 25, 2015 at 3:37 PM, Jessica Forrester <notifications@github.com
> wrote:

> [merge]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4317#issuecomment-134696146>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

**Comment by jwforres** [comment]
Test flake [merge]

**Comment by liggitt** [comment]
for posterity:
```
1)  authenticated e2e-user when creating a new project should successfully create a new project
  - Expected false to be true.
      at /data/src/github.com/openshift/origin/assets/test/e2e/test.js:303:91
```

**Comment by spadgett** [comment]
> Expected false to be true.

Words to meditate on.

**Comment by liggitt** [comment]
Get used to disappointment

**Comment by smarterclayton** [comment]
Expected false to be FILENOTFOUND

On Wed, Aug 26, 2015 at 11:59 AM, Jordan Liggitt <notifications@github.com>
wrote:

> Get used to disappointment
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4317#issuecomment-135050505>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

**Comment by jwforres** [comment]
@spadgett the test case failure is because you are hiding the project h1 now when the project is empty

https://github.com/openshift/origin/blob/master/assets/test/e2e/test.js#L303

**Comment by spadgett** [comment]
@jwforres Thanks, I removed that line since the `h1` is no longer there. I'm not sure if `goToAddToProjectPage` will fail because we added the icons.

[test]

**Comment by jwforres** [comment]
Hmm instead of removing it can you change it to check for the project name
in the project selector dropdown?

**Comment by spadgett** [comment]
@jwforres I can give it a shot, but the e2e tests don't run reliably for me in Vagrant. Is this the right command?

```
TEST_ASSETS=true TEST_ASSETS_HEADLESS=true hack/test-end-to-end.sh
```

 Is there a way to run just the UI tests?

**Comment by jwforres** [comment]
On Wed, Aug 26, 2015 at 2:03 PM, Sam Padgett <notifications@github.com>
wrote:

> @jwforres <https://github.com/jwforres> I can give it a shot, but the e2e
> tests don't run reliably for me in Vagrant. Is this the right command?
>
> TEST_ASSETS=true TEST_ASSETS_HEADLESS=true hack/test-end-to-end.sh
>
> Yes.  Although I don't know if the latest vbox has been updated to include
chrome.  Have you tried launching an ec2 instance instead, syncing your
source and then running the tests? Its a bit slow rebuilding everything,
but should work.

>  Is there a way to run just the UI tests?
>

Unfortunately not today @liggitt started to look at whether it would be
possible.  Right now the other end to end tests are populating the empty
state of the project for the first test.

**Comment by spadgett** [comment]
[test]

**Comment by spadgett** [comment]
@jwforres I updated the test. Please take a look.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**